rts/Lua/LuaSyncedCtrl.cpp

```diff
@@ -917,6 +920,43 @@ int LuaSyncedCtrl::SetAllyTeamStartBox(lua_State* L)
 	return 0;
 }
 
+/*** Set the starting position of a team.
+ * 
+ * @function Spring.SetTeamStartPosition
+ * @param teamID integer
+ * @param x number left position (elmos)
+ * @param y number vertical position (elmos)
+ * @param z number top position (elmos)
+ * @return boolean true if the position was set, false if the teamID is invalid or if engine start position selection is enabled.
+ */
+int LuaSyncedCtrl::SetTeamStartPosition(lua_State* L)
+{
+	if (modInfo.useStartPositionSelecter == true) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
+	const unsigned int teamID = luaL_checkint(L, 1);
+	const float x = luaL_checkfloat(L, 2);
+	const float y = luaL_checkfloat(L, 3);
+	const float z = luaL_checkfloat(L, 4);
+
+	if (!teamHandler.IsValidTeam(teamID)) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
+	// Set the position locally
+	teamHandler.SetTeamStartPosition(teamID, float3(x, y, z));
+
+	// Using PLAYER_RDYSTATE_UPDATED to indicate position change without affecting ready state
+	if (clientNet != nullptr) {
+		clientNet->Send(CBaseNetProtocol::Get().SendStartPos(gu->myPlayerNum, teamID, CPlayer::PLAYER_RDYSTATE_UPDATED, x, y, z));
+	}
```

Comment:
This isn't allowed in synced but also shouldn't be needed. It sounds good for a separate "RequestStartPos" function in UnsyncedCtrl though (there it would be without the `teamhandler.Set` part).

Reply:
It didn't work without this clientNet->Send call. It was because the BAR game start widget was expecting a `gadget:AllowStartPosition` callin that handled the startPositions. I managed to improve the code to also trigger the callin (the previous "fix" also called this AllowStartPosition but in the incorrect way indeed).

Reply:
Try this and see if it helps with the `clientNet->Send` removed: https://github.com/sprunk/Beyond-All-Reason/commit/949c46a253139d710b84bc87689bbe48d8804062

Reply:
Thanks! I've fixed it by making sure to call `eventHandler.AllowStartPosition` before setting the position.